src/sage/schemes/projective/projective_space.py

```diff
@@ -2266,6 +2267,71 @@ def line_through(self, p, q):
         m = matrix(3, list(self.gens()) + list(p) + list(q))
         return Curve([f for f in m.minors(3) if f])
 
+    def coherent_sheaf(self, module, twist=0):
+        r"""
+        Return the sheaf defined by the graded ``module``.
+
+        If ``twist`` is a non-zero integer `n`, the sheaf twisted by
+        `\OO_{\PP^r}(n)` is returned.
+
+        INPUT:
+
+        - ``module`` -- a free module or a quotient module over the coordinate
+          ring of this space
+
+        - ``twist`` -- (default: `0`) an integer
+
+        EXAMPLES::
+
+            sage: P2 = ProjectiveSpace(QQ, 2, 'x')
+            sage: S = P2.coordinate_ring()
+            sage: SS = FreeModule(S, 2)
+            sage: P2.coherent_sheaf(SS, twist=2)
+            Twisted coherent sheaf on Projective Space of dimension 2 over Rational Field
+        """
+        from sage.schemes.projective.coherent_sheaf import CoherentSheaf_on_projective_space
+        return CoherentSheaf_on_projective_space(self, module, twist=twist)
+
+    def structure_sheaf(self, twist=0):
+        r"""
+        Return the structure sheaf `\OO_{\PP^r}` of this projective space.
+
+        If ``twist`` is a non-zero integer `n`, the sheaf twisted by
+        `\OO_{\PP^r}(n)` is returned.
+
+        INPUT:
+
+        - ``twist`` -- (default: `0`) an integer
+
+        EXAMPLES::
+
+            sage: P3.<x0,x1,x2,x3> = ProjectiveSpace(QQ, 3)
+            sage: P3.structure_sheaf()
+            Coherent sheaf on Projective Space of dimension 3 over Rational Field
+        """
+        M = FreeModule(self.coordinate_ring(), rank=1)
+        return self.coherent_sheaf(M, twist=twist)
+
+    def arithmetic_genus(self):
+        r"""
+        Return the arithmetic genus of this projective space.
+
+        This is known to be zero. This method simply checks the fact.
```

Comment:
I think it would be better to simply return `0` here. Then you can add doctests that verifies it by the computation.

Reply:
OK. Done.